nomad/structs/services.go

```diff
@@ -781,6 +781,18 @@ func (s *Service) MakeUniqueIdentityName() string {
 	return fmt.Sprintf("%v-%v", s.Name, s.PortLabel)
 }
 
+// IdentityHandle returns a WorkloadIdentityHandle which is a pair of service
+// identity name and service name.
+func (s *Service) IdentityHandle() *WIHandle {
+	if s.Identity != nil {
+		return &WIHandle{
+			IdentityName:       s.Identity.Name,
+			WorkloadIdentifier: s.Name,
```

Comment:
should this perhaps be `s.MakeUniqueIdentityName`?

Reply:
actually yes, I think it should be. And every time we insert any `WIHandle`s, we'd call this method.